pygfx/renderers/wgpu/engine/shared.py

```diff
@@ -130,6 +142,86 @@ def _create_diagnostics(self):
         PygfxResourceDiagnostics("pygfx_resources")
 
 
+def select_power_preference(power_preference):
+    """Select whether a powerful or battery-friendly GPU is selected.
+
+    Accepts a value from ``wgpu.PowerPreference``: "high-performance" or "low-power".
+
+    This function must be called before before the first ``Renderer`` is created.
+    """
+    if power_preference not in wgpu.PowerPreference:
+        raise ValueError(
+            f"select_power_preference() received invalid value for {repr(wgpu.PowerPreference)}."
+        )
+    if Shared._instance is not None:
+        raise RuntimeError(
+            "The select_power_preference() function must be called before creating the first renderer."
+        )
+    Shared._power_preference = power_preference
+
+
+def select_adapter(adapter):
+    """Select a specific adapter / GPU.
+
+    Select an adapter as obtained via ``wgpu.gpu.enumerate_adapters()``, which
+    can be useful in multi-gpu environments.
+
+    For example::
+
+        adapters = wgpu.gpu.enumerate_adapters()
+        adapters_tesla = [a for a in adapters if "Tesla" in a.summary]
+        adapters_discrete = [a for a in adapters if "DiscreteGPU" in a.summary]
+        pygfx.renderers.wgpu.select_adapter(adapters_discrete[0])
```

Comment:
I guess this isn't exposed at top level in case other renderers are added in the future?

Reply:
Yes, the `pygfx.renderers.wgpu` is a public namespace for power-users.  It has a special place in the docs: https://pygfx.readthedocs.io/en/latest/advanced_api.html The `print_wgpu_report()` function is one exception that is promoted to the root namespace.

Reply:
We could promote other functions to the root namespace as well if there is a demand for that. But I'm a bit careful doing that since its much easier to add api then to remove it (and pissing off users).

Reply:
makes sense, people who want to select a GPU will probably dig and find it easy enough

Reply:
And the example will help 😄